org.wso2.carbon.identity.dao.stored.procedure/src/main/java/org/wso2/carbon/extension/identity/dao/stored/procedure/internal/StoreProcedureBasedDAOComponent.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.extension.identity.dao.stored.procedure.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.extension.identity.dao.stored.procedure.StoreProcedureBasedScopeClaimMappingDAOImpl;
+import org.wso2.carbon.identity.core.persistence.IdentityDBInitializer;
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.openidconnect.dao.ScopeClaimMappingDAO;
+
+import java.sql.Connection;
+import java.util.Map;
+
+/**
+ * The component for the Store Procedure Based DAO.
+ */
+@Component(
+        name = "extension.identity.dao.stored.procedure",
+        immediate = true
+)
+public class StoreProcedureBasedDAOComponent {
+
+    private static final Log log = LogFactory.getLog(StoreProcedureBasedDAOComponent.class);
+    private static final String OIDC_SCOPE_CLAIM_MAPPING_DAO_NAME = "oidc_scope_claim_mapping";
+    private static final String MSSQL = "mssql";
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+
+        Map<String, Boolean> storeProcedureBasedDAOConfig = IdentityUtil.getStoreProcedureBasedDAOConfigurationHolder();
+        Boolean scopeClaimMappingDAOEnabled = storeProcedureBasedDAOConfig.get(OIDC_SCOPE_CLAIM_MAPPING_DAO_NAME);
+        if (scopeClaimMappingDAOEnabled) {
+            Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+            try {
+                String databaseType = IdentityDBInitializer.getDatabaseType(dbConnection);
+                if (MSSQL.equals(databaseType)) {
```

Comment:
Also we need a config to enable this, as this feature need to be enabled per each environment after creating the sp

Reply:
As one deployment will use only one Database type, this feature can be enabled using the following config.
```
[stored_procedure_dao]
oidc_scope_claim_mapping = true
```

This if the condition is to validate whether if someone using MySQL or other unsupported database enable this config by mistake. In that case, we will add a warn log and skip registering this stored procedure based DAO implementation.

Reply:
The feature can be enabled per each environment after creating the sp using the following config,
```
[stored_procedure_dao]
oidc_scope_claim_mapping = true
```

-----
This `if (MSSQL.equals(databaseType))` validation is to add a warn log and make this config redundant if someone enables this feature for non-supported Database type (To support new Database type, we need to have code level changes.).